services/src/pro/contexts/mod.rs

```diff
@@ -24,3 +37,202 @@ pub trait ProContext: Context<Session = UserSession> {
     fn user_db_ref(&self) -> &Self::UserDB;
     fn oidc_request_db(&self) -> Option<&OidcRequestDb>;
 }
+
+pub struct ExecutionContextImpl<S, D, L>
+where
+    D: DatasetDb<S>,
+    L: LayerProviderDb,
+    S: Session,
+{
+    dataset_db: Arc<D>,
+    layer_provider_db: Arc<L>,
+    thread_pool: Arc<ThreadPool>,
+    session: S,
+    tiling_specification: TilingSpecification,
+}
```

Comment:
Why did you need to create that?

Reply:
the execution context is now responsible for wrapping the operators with a statistics operator. As it should only do so in the pro version, we need a separate ExecutionContextImpl for Pro. Previously we used the same in pro and free.